src/Swarm/Util.hs

```diff
@@ -239,6 +240,23 @@ applyWhen False _ x = x
 hoistMaybe :: (Applicative m) => Maybe b -> MaybeT m b
 hoistMaybe = MaybeT . pure
 
+-- | Like 'unsnoc', but for 'NonEmpty' so without the 'Maybe'
+--
+-- Taken from Cabal-syntax Distribution.Utils.Generic.
+--
+-- Example:
+--
+-- >>> unsnocNE (1 :| [2, 3])
+-- ([1,2],3)
+--
+-- >>> unsnocNE (1 :| [])
+-- ([],1)
+unsnocNE :: NonEmpty a -> ([a], a)
+unsnocNE (x :| xs) = go x xs
```

Comment:
@kostmo I noticed there is a new HLint warning, but I could not find how to do this using our libraries.

I found this using Hoogle and borrowed the code.